File: clients/health-android/src/main/kotlin/io/zen70/healthgateway/GatewayBootstrapConfig.kt
package io.zen70.healthgateway

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Bootstrap configuration received from the gateway bootstrap receipt endpoint.
 *
 * Serializes to/from the JSON contract emitted by backend/api/nodes.py
 * (NodeBootstrapReceipt). Field names match the backend snake_case convention.
 *
 * Security: [nodeToken] must be stored in the Android Keystore
 * (via EncryptedSharedPreferences or BiometricPrompt-wrapped keystore entry),
 * never in plain SharedPreferences or disk. [gatewayCaFile] must be loaded
 * as a custom X509TrustManager in OkHttpClient to enforce certificate pinning
 * — see GatewayHttpClient for the TLS pinning implementation.
 */
@Serializable
data class GatewayBootstrapConfig(
    @SerialName("tenant_id") val tenantId: String,
    @SerialName("node_id") val nodeId: String,
    @SerialName("node_token") val nodeToken: String,
    @SerialName("gateway_base_url") val gatewayBaseUrl: String,
    @SerialName("gateway_ca_file") val gatewayCaFile: String,
)

File: clients/health-ios/Sources/HealthGatewayClient/GatewayBootstrapConfig.swift
import Foundation

public struct GatewayBootstrapConfig: Sendable, Equatable, Codable {
    public let tenantID: String
    public let nodeID: String
    public let nodeToken: String
    public let gatewayBaseURL: URL
    public let gatewayCAFile: String

    public init(
        tenantID: String,
        nodeID: String,
        nodeToken: String,
        gatewayBaseURL: URL,
        gatewayCAFile: String
    ) {
        self.tenantID = tenantID
        self.nodeID = nodeID
        self.nodeToken = nodeToken
        self.gatewayBaseURL = gatewayBaseURL
        self.gatewayCAFile = gatewayCAFile
    }

    // Maps Swift camelCase properties to the snake_case JSON keys emitted by
    // the backend bootstrap receipt (backend/api/nodes.py NodeBootstrapReceipt).
    enum CodingKeys: String, CodingKey {
        case tenantID = "tenant_id"
        case nodeID = "node_id"
        case nodeToken = "node_token"
        case gatewayBaseURL = "gateway_base_url"
        case gatewayCAFile = "gateway_ca_file"
    }
}
